.circleci/shellcheck.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+#!/usr/bin/env bash
+# Sets some Bash options to encourage well formed code.
+# For example, some of the options here will cause the script to terminate as
+# soon as a command fails. Another option will cause an error if an undefined
+# variable is used.
+# See: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html
+
+# Any trap on ERR is inherited by shell functions, command substitutions, and
+# commands executed in a subshell environment. The ERR trap is normally not
+# inherited in such cases.
+set -o errtrace
+
+# Any trap on DEBUG and RETURN are inherited by shell functions, command
+# substitutions, and commands executed in a subshell environment. The DEBUG and
+# RETURN traps are normally not inherited in such cases.
+set -o functrace
+
+# Exit if any command exits with a non-zero exit status.
+set -o errexit
+
+# Exit if script uses undefined variables.
+set -o nounset
+
+# Prevent masking an error in a pipeline.
+# Look at the end of the 'Use set -e' section for an excellent explanation.
+# see: https://www.davidpashley.com/articles/writing-robust-shell-scripts/
+set -o pipefail
+
+# Less eager word splitting - no space.
+IFS=$'\n\t'
```

Comment:
By default, IFS contains a space, newline and tab char. Leaving the space means that, in some situations, text with spaces in them will be split...which is _generally_ not the case. I generally leave this as a default because I usually find myself removing the space in IFS for most cases but this can be removed. I guess it's more of a personal preference. Let me know if you want me to remove it.

Reply:
I worry that there may be code relying on spaces being part of `IFS` and that this would be a very subtle backwards incompatibility that wreaks havoc... Unless there's a very compelling reason to keep it, I'd be tempted to leave `IFS` at its default value.

Reply:
Can do! I'll take it out although I do see this convention used more often than not.